src/pydot/core.py

```diff
@@ -314,8 +314,12 @@ def needs_quotes(s):
     if s in dot_keywords:
         return False
 
-    chars = [ord(c) for c in s if ord(c) > 0x7F or ord(c) == 0]
-    if chars and not id_re_dbl_quoted.match(s) and not id_re_html.match(s):
+    has_high_chars = any([ord(c) > 0x7F or ord(c) == 0 for c in s])
+    if (
+        has_high_chars
+        and not id_re_dbl_quoted.match(s)
+        and not id_re_html.match(s)
+    ):
```

Comment:
Not strictly necessary as part of this change, but IMHO more readable, and definitely more efficient. Instead of storing a list of the ordinal value for every non-ASCII character in the string, just store a single boolean indicating whether it contains any non-ASCII characters (since that's all we care about).